akka-projection-dynamodb/src/main/scala/akka/projection/dynamodb/internal/DynamoDBOffsetStore.scala

```diff
@@ -248,9 +251,20 @@ private[projection] class DynamoDBOffsetStore(
   private def readTimestampOffset(): Future[TimestampOffsetBySlice] = {
     val oldState = state.get()
     // retrieve latest timestamp for each slice, and use the earliest
-    val futTimestamps =
-      (minSlice to maxSlice).map(slice => dao.loadTimestampOffset(slice).map(optTimestamp => slice -> optTimestamp))
-    val offsetBySliceFut = Future.sequence(futTimestamps).map(_.collect { case (slice, Some(ts)) => slice -> ts }.toMap)
+    val offsetBySliceFut =
+      Source(minSlice to maxSlice)
+        .mapAsyncUnordered(settings.offsetSliceReadParallelism) { slice =>
+          dao
+            .loadTimestampOffset(slice)
+            .map { optTimestampOffset =>
+              optTimestampOffset.map { timestampOffset => slice -> timestampOffset }
+            }(ExecutionContext.parasitic)
+        }
+        .mapConcat(identity)
+        .runWith(Sink.fold(Map.empty[Int, TimestampOffset]) { (offsetMap, sliceAndOffset: (Int, TimestampOffset)) =>
+          offsetMap + sliceAndOffset
+        })(matFromSystem(system))
```

Comment:
maybe replace `matFromSystem` with `implicit val sys = system`, just to make it more like normal users would do it?